src/metatrain/utils/io.py

```diff
@@ -156,21 +156,47 @@ def load_model(
     if is_exported_file(path):
         return load_atomistic_model(path, extensions_directory=extensions_directory)
     else:  # model is a checkpoint
-        checkpoint = torch.load(path, weights_only=False, map_location="cpu")
-
-        architecture_name = checkpoint["architecture_name"]
-        if architecture_name not in find_all_architectures():
-            raise ValueError(
-                f"Checkpoint architecture '{architecture_name}' not found "
-                "in the available architectures. Available architectures are: "
-                f"{find_all_architectures()}"
-            )
-        architecture = import_architecture(architecture_name)
-
-        try:
-            return architecture.__model__.load_checkpoint(path)
-        except Exception as err:
-            raise ValueError(
-                f"path '{path}' is not a valid checkpoint for the {architecture_name} "
-                "architecture"
-            ) from err
+        return model_from_checkpoint(path)
+
+
+def model_from_checkpoint(path: Union[str, Path]) -> Any:
+    """TODO"""
```

Comment:
Maybe a little docstring for us :-)

I think adding explicit tests for these functions we can add later or now if you like.